File: Data Structures & Algorithms/permutations/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    permute(nums) {
        let res = [];
        this.backtrack(0, nums, res);
        return res;
    }

    backtrack(index, nums, res){
        if (index === nums.length){
            res.push([...nums]);
            return;
        }

        for (let i = index; i < nums.length; i++){
            [nums[index], nums[i]] = [nums[i], nums[index]];
            this.backtrack(index + 1, nums, res);
            [nums[index], nums[i]] = [nums[i], nums[index]];
        }

    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int n = nums.Length;
        int[] res = new int[n];

        for (int i = 0; i < n; i++){
            int temp = 1;
            for (int j = 0; j < n; j++){
                if (i != j) temp *= nums[j];
            }
            res[i] = temp;
        }

        return res;
    }
}
